sodasql/scan/scan_builder.py

```diff
@@ -101,10 +101,12 @@ def _build_warehouse_yml(self):
 
         if self.warehouse_yml_dict and not self.warehouse_yml:
             from sodasql.scan.warehouse_yml_parser import WarehouseYmlParser
-            warehouse_parser = WarehouseYmlParser(self.warehouse_yml_dict, self.warehouse_yml_file)
-            warehouse_parser.log()
-            self.parsers.append(warehouse_parser)
-            self.warehouse_yml = warehouse_parser.warehouse_yml
+            self.parse_warehouse_yml(WarehouseYmlParser(self.warehouse_yml_dict, self.warehouse_yml_file))
+
+    def parse_warehouse_yml(self, warehouse_parser):
```

Comment:
If I understand correctly, the point is to be able to set the `warehouse_yml` by calling this function from soda_scanner?

Reply:
Yes! As-is currently I can't change it. Alternative is to extract this entire code outside of the scan builder, but then we might end up with a more fragile setup with duplicated code so I just did a small Extract Method refactoring